OGA/RG-351P fix.

SDL 1.2's kmsdrm input code will go out of bound if keycodes higher than BTN_MIDDLE are used.
This is why pressing BTN_DPAD_UP might not do anything but pressing A will for example.
This is also why pressing POWER button did work properly unlike the other buttons
as it was mapped and within the array.

So all we need to do is to map those keys to OpenDingux like buttons
for compatibility with such software.

This will only affect SDL 1.2, which is ultimately what we want it to do
but we could also add an optional patch for SDL2 that could map those
or unmap those etc...

This doesn't fix the DPAD being mapped as an analog stick, we
would have to find another way around that...

--- a/src/video/kmsdrm/SDL_kmsdrmevents.c
+++ b/src/video/kmsdrm/SDL_kmsdrmevents.c
@@ -263,7 +263,7 @@
 	[KEY_INSERT] = SDLK_INSERT,
 	[KEY_DELETE] = SDLK_DELETE,
 
-	[KEY_POWER] = SDLK_POWER,
+	[KEY_POWER] = SDLK_HOME,
 	[KEY_KPEQUAL] = SDLK_KP_EQUALS,
 	[KEY_PAUSE] = SDLK_PAUSE,
 
@@ -285,6 +285,22 @@
 	[BTN_RIGHT] = SDL_BUTTON_RIGHT,
 	[BTN_MIDDLE] = SDL_BUTTON_MIDDLE,
 	
+	[BTN_DPAD_UP] = SDLK_UP,
+	[BTN_DPAD_DOWN] = SDLK_DOWN,
+	[BTN_DPAD_LEFT] = SDLK_LEFT,
+	[BTN_DPAD_RIGHT] = SDLK_RIGHT,
+	[304] = SDLK_LCTRL,
+	[305] = SDLK_LALT,
+	[306] = SDLK_LSHIFT,
+	[307] = SDLK_SPACE,
+	[311] = SDLK_ESCAPE,
+	[310] = SDLK_RETURN,
+	[308] = SDLK_TAB,
+	[309] = SDLK_BACKSPACE,
+	[314] = SDLK_PAGEUP,
+	[315] = SDLK_PAGEDOWN,
+	[312] = SDLK_KP_DIVIDE,
+	[313] = SDLK_KP_PERIOD,
 };
 
 void KMSDRM_InitOSKeymap(_THIS)
